pkg/reconciler/testing/v1/roles.go

```diff
@@ -21,17 +21,53 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// RoleOption enables further configuration of a Role.
+type RoleOption func(*v1.Role)
+type PolicyRuleOption func(*v1.PolicyRule)
+
 // createRole creates a Role in the Kubernetes cluster.
-func CreateRole(namespace, roleName, verbs, resources, apiGroup string) *v1.Role {
-	return &v1.Role{
+func CreateRole(name, namespace string, o ...RoleOption) *v1.Role {
+	r := &v1.Role{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      roleName,
+			Name:      name,
 			Namespace: namespace,
 		},
-		Rules: []v1.PolicyRule{{
-			APIGroups: []string{apiGroup},
-			Resources: []string{resources},
-			Verbs:     []string{verbs},
-		}},
+		Rules: []v1.PolicyRule{},
+	}
+	for _, opt := range o {
+		opt(r)
+	}
+	return r
+}
+
+func WithRoleRules(rules ...v1.PolicyRule) RoleOption {
+	return func(r *v1.Role) {
+		r.Rules = rules
+	}
+}
```

Comment:
```suggestion
func WithRoleRules(rules ...*v1.PolicyRule) RoleOption {
	return func(r *v1.Role) {
		r.Rules = rules
	}
}
```
Maybe here we can take in an array of pointers, so that we don't need to dereference the pointers in the actual testing code? Instead we could do something like:
```go
roleRules := make([]v1.PolicyRule, 0, len(rules))
for _, rule := range rules {
    roleRules = append(roleRules, *rule)
}
```
And then we could set the rules as you have above